### ISSUE TITLE
Distinguish between key value pairs and loops in STAR files

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#463]
Currently we represent STAR files as `OrderedDict`s of pandas `DataFrame` objects. The implementation, when reading a STAR file, treats a regular set of key-value pairs, and a single-line loop the same way, mapping to a single-row `DataFrame`. Ideally we should be able to distinguish between these in our representation.

The key-value form in a STAR file looks like:
```
data_
_field1    val2 
_field2    val2
...
```

While a single row loop looks like:
```
data_
loop_
_field1 
_field2
_field3
val1     val2     val3
```

The code handling the reading and writing of files is in `aspire.storage.starfile` and uses the external `gemmi` library.

The goal would be to alter how the `gemmi.Document` representation of the StarFile is converted in our code. `gemmi` itself does make this distinction with `Loop` and `Pair` objects.

https://gemmi.readthedocs.io/en/latest/cif.html
https://project-gemmi.github.io/python-api/gemmi.cif.html

**Comment by chris-langfield** [comment]
As mentioned in #469 this is of continuing concern with regard to compatibility with other tools. Relion seems to have a robust STAR file reading system, that can handle data being represented as a key value pair or a loop.

**Comment by chris-langfield** [comment]
Closing, addressed in #493
